skglm/experimental/reweighted.py

```diff
@@ -0,0 +1,107 @@
+import numpy as np
+from numpy.linalg import norm
+from skglm.estimators import Lasso
+
+
+def _reweight_by_coefficient_norm(coef):
+    nrm = np.sqrt(norm(coef))
+    return 1 / (2 * nrm + np.finfo(float).eps)
+
+
+class ReweightedLasso(Lasso):
+    r"""Reweighted Lasso estimator.
+
+    The objective reads::
+        ||y-Xw||_2^2 / (2 * n_samples) + \lambda \sqrt{||w||}
+
+    This estimator iteratively solves a non-convex objective by iteratively solving
+    convex surrogates. They are obtained by iteratively re-weighting the \ell_1-norm
+    by a penalty applied to the coefficients.
+
+    Parameters
+    ----------
+    alpha : float, optional
+        Penalty strength.
+
+    n_reweights : int, optional
+        Number of reweighting iterations.
+
+    args: list
+        Parameters for the inner Lasso estimator.
+        Note that `warm_start` is consistently set to `True` for performance reasons.
+
+    kwargs: dict
+        Keyword parameters for the inner Lasso estimator.
+
+    Attributes
+    ----------
+    coef_ : array, shape (n_features,)
+        parameter vector (w in the cost function formula)
+
+    loss_history_ : list
+        Objective history after every reweighting iteration
+
+    References
+    ----------
+    .. [1] Candès et al. (2007), Enhancing sparsity by reweighted l1 minimization
+           https://web.stanford.edu/~boyd/papers/pdf/rwl1.pdf
+    """
+
+    def __init__(self, alpha=1., n_reweights=5, *args, **kwargs):
+        # Warm start is crucial to ensure the solution of surrogate problem i
```

Comment:
warm start is not related to the fact that we solve n_rewieghting problem IMO.

Instead warm start means coef_ is reused at each call to fit.